src/DynamoCore/Library/LibraryServices.cs

```diff
@@ -627,7 +627,10 @@ internal bool LoadNodeLibrary(string library, bool isExplicitlyImportedLib)
                     throwOnFailure: !isExplicitlyImportedLib));
                 return false;
             }
-            importedLibraries.Add(library);
+            if (!importedLibraries.Contains(library))
```

Comment:
so are these 2 if statements mostly related to stoping duplicate nodes from being added to the library?

Reply:
Not really, duplicates are not added to the library anymore because I'm not reloading existing packages unless necessary. I saw that these checks were still needed as there were cases (I can't remember now) where duplicate entries were being added to these lists inadvertently, which were harmless but confusing nonetheless.